File: SozlukJava/app/src/main/java/com/example/sozlukjava/view/EntryDetayFragment.java
package com.example.sozlukjava.view;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.sozlukjava.R;
import com.example.sozlukjava.databinding.FragmentEntryDetayBinding;
import com.example.sozlukjava.model.Comment;
import com.example.sozlukjava.util.SessionManager;
import com.example.sozlukjava.viewmodel.EntryDetayViewModel;

// EntryDetayFragment.java
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import java.util.ArrayList;
import java.util.List;

public class EntryDetayFragment extends Fragment {
    private FragmentEntryDetayBinding binding;

    private EntryDetayViewModel viewModel;
    private SessionManager session;
    private ArrayAdapter<String> adapter;

    private int entryId = -1;
    private List<Comment> commentList = new ArrayList<>();

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = FragmentEntryDetayBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        session = new SessionManager(requireContext());
        viewModel = new ViewModelProvider(this).get(EntryDetayViewModel.class);

        entryId = getArguments() != null ? getArguments().getInt("entryId", -1) : -1;
        if (entryId == -1) return;

        viewModel.loadEntry(entryId);

        viewModel.getEntry().observe(getViewLifecycleOwner(), e -> {
            if (e == null) return;
            binding.tvEntryTitle.setText(e.getTitle());
            binding.tvEntryContent.setText(e.getContent());
            binding.tvEntryAuthor.setText(e.getUsername());
            String created = e.getCreated_at();
            binding.tvEntryDate.setText(created != null && created.length() >= 10 ? created.substring(0, 10) : "");
        });

        // Yorumları yükle
        viewModel.loadComments(entryId);

        viewModel.getComments().observe(getViewLifecycleOwner(), list -> {
            commentList = list != null ? list : new ArrayList<>();
            ArrayList<String> strings = new ArrayList<>();
            for (Comment c : commentList) {
                strings.add(c.getUsername() + ":\n" + c.getComment_text() +
                        "\n👍" + c.getLikes() + " 👎" + c.getDislikes());
            }
            adapter = new ArrayAdapter<>(requireContext(),
                    android.R.layout.simple_list_item_1, strings);
            binding.commentListView.setAdapter(adapter);
        });

        binding.btnYorumGonder.setOnClickListener(v -> {
            String commentText = binding.etComment.getText().toString();
            int userId = session.getUserId();
            if (commentText != null && !commentText.trim().isEmpty()) {
                viewModel.addComment(entryId, userId, commentText);
                binding.etComment.getText().clear();
            }
        });

        binding.commentListView.setOnItemClickListener((parent, v1, position, id) -> {
            Comment comment = commentList.get(position);
            showLikeDislikeDialog(comment);
        });
    }

    private void showLikeDislikeDialog(Comment comment) {
        int userId = session.getUserId();
        String[] options = new String[]{"👍 Beğen", "👎 Beğenme"};
        new AlertDialog.Builder(requireContext())
                .setTitle("Yorumu Oyla")
                .setItems(options, (dialog, which) -> {
                    if (which == 0) {
                        viewModel.voteComment(entryId, comment.getId(), userId, true);
                    } else if (which == 1) {
                        viewModel.voteComment(entryId, comment.getId(), userId, false);
                    }
                })
                .setNegativeButton("İptal", null)
                .show();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: SozlukKotlin/app/src/main/java/com/example/sozlukkotlin/view/EntryDetayFragment.kt
package com.example.sozlukkotlin.view

import android.app.AlertDialog
import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.lifecycleScope
import com.example.sozlukkotlin.R
import com.example.sozlukkotlin.databinding.FragmentEntryDetayBinding
import com.example.sozlukkotlin.model.Comment
import com.example.sozlukkotlin.util.SessionManager
import com.example.sozlukkotlin.viewmodel.EntryDetayViewModel
import kotlinx.coroutines.flow.launchIn
import kotlinx.coroutines.flow.onEach



class EntryDetayFragment : Fragment() {
    private var _binding: FragmentEntryDetayBinding? = null
    private val binding get() = _binding!!

    private lateinit var viewModel: EntryDetayViewModel
    private lateinit var session: SessionManager
    private lateinit var adapter: ArrayAdapter<String>

    private var entryId: Int = -1
    private var commentList: List<Comment> = listOf()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        _binding = FragmentEntryDetayBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        session = SessionManager(requireContext())
        viewModel = ViewModelProvider(this)[EntryDetayViewModel::class.java]

        entryId = arguments?.getInt("entryId") ?: -1
        if (entryId == -1) return

        viewModel.loadEntry(entryId)

        viewModel.entry.onEach { e ->
            e ?: return@onEach
            binding.tvEntryTitle.text = e.title
            binding.tvEntryContent.text = e.content
            binding.tvEntryAuthor.text = e.username
            binding.tvEntryDate.text = e.created_at.take(10)
        }.launchIn(viewLifecycleOwner.lifecycleScope)

        // Yorumları yükle
        viewModel.loadComments(entryId)

        viewModel.comments.onEach { list ->
            commentList = list
            val strings = list.map {
                "${it.username}:\n${it.comment_text}\n👍${it.likes} 👎${it.dislikes}"
            }
            adapter = ArrayAdapter(requireContext(), android.R.layout.simple_list_item_1, strings)
            binding.commentListView.adapter = adapter
        }.launchIn(viewLifecycleOwner.lifecycleScope)

        binding.btnYorumGonder.setOnClickListener {
            val commentText = binding.etComment.text.toString()
            val userId = session.getUserId()
            if (commentText.isNotBlank()) {
                viewModel.addComment(entryId, userId, commentText)
                binding.etComment.text.clear()
            }
        }

        binding.commentListView.setOnItemClickListener { _, _, position, _ ->
            val comment = commentList[position]
            showLikeDislikeDialog(comment)
        }
    }

    private fun showLikeDislikeDialog(comment: Comment) {
        val userId = session.getUserId()
        val options = arrayOf("👍 Beğen", "👎 Beğenme")
        AlertDialog.Builder(requireContext())
            .setTitle("Yorumu Oyla")
            .setItems(options) { _, which ->
                when (which) {
                    0 -> viewModel.voteComment(entryId, comment.id, userId, true)
                    1 -> viewModel.voteComment(entryId, comment.id, userId, false)
                }
            }
            .setNegativeButton("İptal", null)
            .show()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
